### PR TITLE
Add unit tests for PhysicsMaterial

### PR DESCRIPTION
**Pull request by D0V4HKIIN**
Unit tests for PhysicsMaterial for tracker #43440

I am somewhat unsure about the relevancy of theses tests but here they are. Mostly testing setters and getters.

### CODE REVIEW

File: tests/scene/test_physics_material.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#ifndef TEST_PHYSICS_MATERIAL_H
+#define TEST_PHYSICS_MATERIAL_H
+
+#include "scene/resources/physics_material.h"
+#include "tests/test_macros.h"
+
+namespace TestPhysics_material {
+
+TEST_CASE("[Physics_material] Defaults") {
+	PhysicsMaterial *physics_material = memnew(PhysicsMaterial);
```
Comment (AThousandShips) [inline tests/scene/test_physics_material.h:40]:
```suggestion
	Ref<PhysicsMaterial> physics_material;
	physics_material.instantiate();
```
For all, this is a RefCounted type so should use `Ref`

File: tests/scene/test_physics_material.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	CHECK(physics_material->get_friction() == 1.);
+	CHECK(physics_material->is_rough() == false);
+	CHECK(physics_material->get_bounce() == 0.);
+	CHECK(physics_material->is_absorbent() == false);
+
+	memdelete(physics_material);
```
Comment (AThousandShips) [inline tests/scene/test_physics_material.h:47]:
```suggestion
```
Also for all

**Review by AThousandShips — DISMISSED** [review]

**Comment by AThousandShips** [comment]
Thank you for your contribution!

These tests look good to me, but can't say if they cover things that we want to test for or not so this will have to be looked over by the physics team to give their approval

File: tests/scene/test_physics_material.h [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+#ifndef TEST_PHYSICS_MATERIAL_H
+#define TEST_PHYSICS_MATERIAL_H
+
+#include "scene/resources/physics_material.h"
+#include "tests/test_macros.h"
+
+namespace TestPhysics_material {
+
+TEST_CASE("[Physics_material] Defaults") {
```
Comment (Calinou) [inline tests/scene/test_physics_material.h:39]:
```suggestion
TEST_CASE("[PhysicsMaterial] Defaults") {
```

File: tests/scene/test_physics_material.h [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	Ref<PhysicsMaterial> physics_material;
+	physics_material.instantiate();
+
+	CHECK(physics_material->get_friction() == 1.);
+	CHECK(physics_material->is_rough() == false);
+	CHECK(physics_material->get_bounce() == 0.);
+	CHECK(physics_material->is_absorbent() == false);
+}
+
+TEST_CASE("[Physics_material] Friction") {
```
Comment (Calinou) [inline tests/scene/test_physics_material.h:49]:
```suggestion
TEST_CASE("[PhysicsMaterial] Friction") {
```

File: tests/scene/test_physics_material.h [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	Ref<PhysicsMaterial> physics_material;
+	physics_material.instantiate();
+
+	real_t friction = 0.314;
+	physics_material->set_friction(friction);
+	CHECK(physics_material->get_friction() == friction);
+}
+
+TEST_CASE("[Physics_material] Rough") {
```
Comment (Calinou) [inline tests/scene/test_physics_material.h:58]:
```suggestion
TEST_CASE("[PhysicsMaterial] Rough") {
```

File: tests/scene/test_physics_material.h [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	CHECK(physics_material->computed_friction() == -friction);
+
+	rough = false;
+	physics_material->set_rough(rough);
+	CHECK(physics_material->is_rough() == rough);
+
+	CHECK(physics_material->computed_friction() == friction);
+}
+
+TEST_CASE("[Physics_material] Bounce") {
```
Comment (Calinou) [inline tests/scene/test_physics_material.h:77]:
```suggestion
TEST_CASE("[PhysicsMaterial] Bounce") {
```

File: tests/scene/test_physics_material.h [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	Ref<PhysicsMaterial> physics_material;
+	physics_material.instantiate();
+
+	real_t bounce = 0.271;
+	physics_material->set_bounce(bounce);
+	CHECK(physics_material->get_bounce() == bounce);
+}
+
+TEST_CASE("[Physics_material] Absorbent") {
```
Comment (Calinou) [inline tests/scene/test_physics_material.h:86]:
```suggestion
TEST_CASE("[PhysicsMaterial] Absorbent") {
```

**Review by rburing — APPROVED** [review]
Looks good to me, as it tests all that can be tested in isolation (using only this class). Please [squash the commits](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) as required by our workflow (and ideally, change the commit message to the current title of this PR).

**Comment by D0V4HKIIN** [comment]
Done!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**